### ISSUE TITLE
fairseq-generate: Dataset not found

### ISSUE THREAD

**Issue by Lucianod28** [pytorch/fairseq#974]
I want to use my model to generate, so I ran:
`fairseq-generate data-bin/train.tokenized.queries-paraphrased --path checkpoints/fconv/checkpoint_best.pt --batch-size 128 --beam 5 --remove-bpe`
and I got the following output:
```
Namespace(beam=5, cpu=False, criterion='cross_entropy', data='data-bin/train.tokenized.queries-paraphrased', dataset_impl='cached', diverse_beam_groups=-1, diverse_beam_strength=0.5, force_anneal=None, fp16=False, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, gen_subset='test', lazy_load=False, left_pad_source='True', left_pad_target='False', lenpen=1, log_format=None, log_interval=1000, lr_scheduler='fixed', lr_shrink=0.1, match_source_len=False, max_len_a=0, max_len_b=200, max_sentences=128, max_source_positions=1024, max_target_positions=1024, max_tokens=None, memory_efficient_fp16=False, min_len=1, min_loss_scale=0.0001, model_overrides='{}', momentum=0.99, nbest=1, no_beamable_mm=False, no_early_stop=False, no_progress_bar=False, no_repeat_ngram_size=0, num_shards=1, num_workers=0, optimizer='nag', path='checkpoints/fconv/checkpoint_best.pt', prefix_size=0, print_alignment=False, quiet=False, raw_text=False, remove_bpe='@@ ', replace_unk=None, required_batch_size_multiple=8, results_path=None, sacrebleu=False, sampling=False, sampling_topk=-1, sampling_topp=-1.0, score_reference=False, seed=1, shard_id=0, skip_invalid_size_inputs_valid_test=False, source_lang=None, target_lang=None, task='translation', tbmf_wrapper=False, temperature=1.0, tensorboard_logdir='', threshold_loss_scale=None, unkpen=0, unnormalized=False, upsample_primary=1, user_dir=None, warmup_updates=0, weight_decay=0.0)
| [queries] dictionary: 9776 types
| [paraphrased] dictionary: 9816 types
Traceback (most recent call last):
  File "/home/luciano/Paraphrasing/venv/bin/fairseq-generate", line 10, in <module>
    sys.exit(cli_main())
  File "/home/luciano/Paraphrasing/venv/lib/python3.5/site-packages/fairseq_cli/generate.py", line 188, in cli_main
    main(args)
  File "/home/luciano/Paraphrasing/venv/lib/python3.5/site-packages/fairseq_cli/generate.py", line 35, in main
    task.load_dataset(args.gen_subset)
  File "/home/luciano/Paraphrasing/venv/lib/python3.5/site-packages/fairseq/tasks/translation.py", line 188, in load_dataset
    max_target_positions=self.args.max_target_positions,
  File "/home/luciano/Paraphrasing/venv/lib/python3.5/site-packages/fairseq/tasks/translation.py", line 48, in load_langpair_dataset
    raise FileNotFoundError('Dataset not found: {} ({})'.format(split, data_path))
FileNotFoundError: Dataset not found: test (data-bin/train.tokenized.queries-paraphrased)
```
but the file is there, `ls -al data-bin`, shows:
```
drwxrwxr-x 2 luciano luciano 4096 Aug  5 09:32 train.tokenized.queries-paraphrased
```
I've also tried using the full path to the directory, `/home/luciano/Paraphrasing/seq2seq/data-bin/train.tokenized.queries-paraphrased`, but got the same error.

I'm running Ubuntu 16.04 with fairseq version 0.7.2.

**Comment by edunov** [comment]
It is looking for "test" subset in this dataset folder. You should have files like test.*.idx and test.*.bin inside the dataset folder,  if you don't check how did you make this dataset.

**Comment by Lucianod28** [comment]
Oh I see I only generated train and valid splits.

**Comment by divyangashah** [comment]
hi, @edunov 
I have files you mentioned like test*.idx & test.*.bin.
still i am facing same issue:

Traceback (most recent call last):
  File "/usr/local/bin/fairseq-generate", line 8, in <module>
Namespace(beam=5, bpe=None, cpu=False, criterion='cross_entropy', data='/content/drive/My Drive/MachineTranslation/FairSeq/data-bin/wmt17.zh-en.newstest2017', dataset_impl=None, decoding_format=None, diverse_beam_groups=-1, diverse_beam_strength=0.5, empty_cache_freq=0, force_anneal=None, fp16=False, fp16_init_scale=128, fp16_scale_tolerance=0.0, fp16_scale_window=None, gen_subset='test', iter_decode_eos_penalty=0.0, iter_decode_force_max_iter=False, iter_decode_max_iter=10, lazy_load=False, left_pad_source='True', left_pad_target='False', lenpen=1, load_alignments=False, log_format=None, log_interval=1000, lr_scheduler='fixed', lr_shrink=0.1, match_source_len=False, max_len_a=0, max_len_b=200, max_sentences=128, max_source_positions=1024, max_target_positions=1024, max_tokens=None, memory_efficient_fp16=False, min_len=1, min_loss_scale=0.0001, model_overrides='{}', momentum=0.99, nbest=1, no_beamable_mm=False, no_early_stop=False, no_progress_bar=False, no_repeat_ngram_size=0, num_shards=1, num_workers=1, optimizer='nag', path='/content/drive/My Drive/MachineTranslation/FairSeq/data-bin/wmt17.zh-en.lightconv-glu/model.pt', prefix_size=0, print_alignment=False, print_step=False, quiet=False, raw_text=False, remove_bpe='@@ ', replace_unk=None, required_batch_size_multiple=8, results_path=None, retain_iter_history=False, sacrebleu=False, sampling=False, sampling_topk=-1, sampling_topp=-1.0, score_reference=False, seed=1, shard_id=0, skip_invalid_size_inputs_valid_test=False, source_lang=None, target_lang=None, task='translation', temperature=1.0, tensorboard_logdir='', threshold_loss_scale=None, tokenizer=None, truncate_source=False, unkpen=0, unnormalized=False, upsample_primary=1, user_dir=None, warmup_updates=0, weight_decay=0.0)
| [zh] dictionary: 48216 types
| [en] dictionary: 32595 types
    sys.exit(cli_main())
  File "/usr/local/lib/python3.6/dist-packages/fairseq_cli/generate.py", line 199, in cli_main
    main(args)
  File "/usr/local/lib/python3.6/dist-packages/fairseq_cli/generate.py", line 33, in main
    task.load_dataset(args.gen_subset)
  File "/usr/local/lib/python3.6/dist-packages/fairseq/tasks/translation.py", line 219, in load_dataset
    truncate_source=self.args.truncate_source,
  File "/usr/local/lib/python3.6/dist-packages/fairseq/tasks/translation.py", line 52, in load_langpair_dataset
    raise FileNotFoundError('Dataset not found: {} ({})'.format(split, data_path))
FileNotFoundError: Dataset not found: test (/content/drive/My Drive/MachineTranslation/FairSeq/data-bin/wmt17.zh-en.newstest2017)

can you point out something?

**Comment by Crista23** [comment]
@divyangashah I get the same error although I have the test files, did you find any solution to this?
